src/lib.rs

```diff
@@ -645,10 +665,10 @@ pub trait Rng: RngCore {
     /// // First meaningful use of `gen_weighted_bool`.
     /// println!("{}", rng.gen_weighted_bool(3));
     /// ```
-    #[deprecated(since="0.5.0", note="use gen_bool instead")]
+    #[deprecated(since="0.5.0", note="use gen_ratio instead")]
     fn gen_weighted_bool(&mut self, n: u32) -> bool {
-        // Short-circuit after `n <= 1` to avoid panic in `gen_range`
-        n <= 1 || self.gen_range(0, n) == 0
+        // Short-circuit after `n <= 1` to avoid panic in `gen_ratio`
```

Comment:
This comment is not correct with the change. But maybe it is best to not change this deprecated method (also it will be removed soon https://github.com/rust-lang-nursery/rand/pull/499).